packages/components/src/Modal/createPortal.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { ReactNode, useEffect } from "react";
+import ReactDOM from "react-dom";
+
+interface CreatePortalProps {
+  children: ReactNode | ReactNode[];
+}
+
+export function CreatePortal({ children }: CreatePortalProps) {
+  return <Portal>{children}</Portal>;
+}
+
+interface PortalProps {
+  children: ReactNode | ReactNode[];
+}
+
+export function Portal({ children }: PortalProps) {
+  const modalRoot = document.body.appendChild(document.createElement("div"));
```

Comment:
I think we can pull this out to the module scope. That way we don't create a new `portalRoot` for each portal.